platform/ios/platform/ios/src/MGLMapView.mm

```diff
@@ -1003,17 +1003,47 @@ - (void)updateViewsPostMapRendering {
     [self processPendingBlocks];
 }
 
-- (void)renderSync
+- (BOOL)renderSync
 {
-    if ( ! self.dormant && _rendererFrontend)
+    BOOL hasPendingBlocks = (self.pendingCompletionBlocks.count > 0);
+
+    if (!self.needsDisplayRefresh && !hasPendingBlocks) {
+        return NO;
+    }
+
+    BOOL needsRender = self.needsDisplayRefresh;
+
+    self.needsDisplayRefresh = NO;
+
+    if (!self.dormant && needsRender)
     {
-        _rendererFrontend->render();
+        // It's important to call this *before* `_rendererFrontend->render()`, as
+        // that function saves the current `updateParameters` before rendering. If this
+        // occurs after then the views will be a frame behind.
+        //
+        // The update parameters will have been updated earlier, for example by
+        // calls to easeTo, flyTo, called from gesture handlers.
+        
+        [self updateViewsWithCurrentUpdateParameters];
+      
+        if (_rendererFrontend) {
+            
+            _rendererFrontend->render();
+
+        }
+        
+    }
+
+    if (hasPendingBlocks) {
```

Comment:
I wonder, is this block needed given updateViewsWithCurrentUpdateParameters invokes processPendingBlocks already? Unless you want to invoke processPendingBlocks when the condition in line 1018 evaluates to false, but in this case, can we drop hasPendingBlocks and have an else block in 1035 instead?

Reply:
This code was taken from https://github.com/mapbox/mapbox-gl-native-ios/blob/ddf8e35970d1a8a46171bd798bcfa8f22d2760e0/platform/ios/src/MGLMapView.mm#L1180 in `mapbox-gl-native-ios`. I did not consider removing that as I would assume it was originally placed there for a reason.

What what I see it needs to be called before and after `_rendererFrontend->render();`.

Reply:
I see that updateViewsWithCurrentUpdateParameters does not contain processPendingBlocks call in the code you posted (mapbox code base: https://github.com/mapbox/mapbox-gl-native-ios/blob/ddf8e35970d1a8a46171bd798bcfa8f22d2760e0/platform/ios/src/MGLMapView.mm#L1180). Removing it would sort out my concern, as it processPendingBlocks would not get called twice in the render method :).

Reply:
@sulewicz Thanks, you were right, that shouldn't have been there. It has been removed. 👍